File: 2.Dagger/Subcomponents/PlusMethod/src/main/java/com/app/subcomponents/MainActivity.java
package com.app.subcomponents;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {
    AppComponent appComponent;
    ActivityComponent activityComponent;

    @Inject
    String str;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        appComponent = DaggerAppComponent.factory().create();
        activityComponent = appComponent.plusActivityComponent().create();

        activityComponent.inject(this);

        Log.i(TAG, "injected String - " + str);
    }

    private static final String TAG = "MainActivity";
}

File: 3.Koin/Component/app/src/main/java/com/app/component/MainActivity.kt
package com.app.component

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import org.koin.core.context.startKoin

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        startKoin {
            modules(dataModule)
        }
        val component = MyComponent()
        Log.i(TAG, "onCreate: ${component.first}")
        Log.i(TAG, "onCreate: ${component.second}")

    }

    companion object {
        private const val TAG = "MainActivity"
    }
}